astronomer/providers/sftp/example_dags/example_sftp.py

```diff
@@ -16,6 +46,168 @@
+
+
+def create_instance_with_security_group() -> None:
+    """Create ec2 instance"""
+    import boto3
+
+    ec2 = boto3.resource("ec2", **AWS_S3_CREDS)
+    instance = ec2.create_instances(
+        ImageId=AMI_ID,
+        MinCount=1,
+        MaxCount=1,
+        InstanceType=SFTP_INSTANCE_TYPE,
+        KeyName=PEM_FILENAME,
+        SecurityGroups=[INBOUND_SECURITY_GROUP],
+    )
+    instance_id = instance[0].id
+    ti = get_current_context()["ti"]
+    ti.xcom_push(key="ec2_instance_id", value=instance_id)
+
+
+def get_ec2_instance_details(task_instance: "TaskInstance") -> None:
+    """Get the EC2 instance details by id retrieved from the Xcom"""
+    import boto3
+
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+    ec2_instance_id_xcom = task_instance.xcom_pull(key="ec2_instance_id", task_ids=["create_ec2_instance"])[0]
+    response = client.describe_instances(
+        InstanceIds=[ec2_instance_id_xcom],
+    )
+    instance_details = response["Reservations"][0]["Instances"][0]
+    ti = get_current_context()["ti"]
+    ti.xcom_push(
+        key="instance_response_master_security_group", value=instance_details["SecurityGroups"][0]["GroupId"]
+    )
+    ti.xcom_push(key="instance_public_dns_name", value=instance_details["PublicDnsName"])
+
+
+def add_inbound_rule_for_security_group(task_instance: "TaskInstance") -> None:
+    """
+    Sets the inbound rule for the aws security group, based on
+    current ip address of the system.
+    """
+    import boto3
+    from botocore.exceptions import ClientError
+
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+    current_docker_ip = get("https://api.ipify.org").text
+
+    # Allow SSH traffic on port 22 and copy file to ec2 instance.
+    try:
+        client.authorize_security_group_ingress(
+            GroupId=task_instance.xcom_pull(
+                key="instance_response_master_security_group", task_ids=["get_instance_details"]
+            )[0],
+            IpPermissions=[
+                {
+                    "IpProtocol": "tcp",
+                    "FromPort": 22,
+                    "ToPort": 22,
+                    "IpRanges": [{"CidrIp": str(current_docker_ip) + "/32"}],
+                }
+            ],
+        )
+    except ClientError as error:
+        if error.response.get("Error", {}).get("Code", "") == BOTO_DUPLICATE_PERMISSION_ERROR:
+            logging.error(
+                "Ingress for port 22 already authorized. Error message is: %s",
+                error.response["Error"]["Message"],
+            )
+        else:
+            raise error
+
+
+def ssh_and_run_command(task_instance: Any, **kwargs: Any) -> None:
+    """
+    Load the private_key from airflow variable and creates a pem_file
+    at /tmp/. SSH into the machine and execute the bash script from the list
+    of commands.
+    """
+    # remove the file if it exists
+    if os.path.exists(f"/tmp/{PEM_FILENAME}.pem"):
+        os.remove(f"/tmp/{PEM_FILENAME}.pem")
```

Comment:
Will this impact running SFTP example dag parallel with Hive or Livy dag since they also create the same location when running on astro cloud?

Reply:
This is scheduled after the hive and livy its not parallel